src/main/java/training/supportbank/FileService.java

```diff
@@ -0,0 +1,53 @@
+package training.supportbank;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class FileService {
+
+    private final String filePath;
+
+    public FileService(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    public Set<String> getAccountNames() {
+        Stream<String[]> stream = parseCSV();
+        Set<String> accountNames = stream
+                .map(line -> line[1])
+                .collect(Collectors.toSet());
+        stream = parseCSV();
+        Set<String> otherAccountNames = stream.map(line -> line[2])
+                .collect(Collectors.toSet());
+        accountNames.addAll(otherAccountNames);
+        return accountNames;
+    }
+
+    public List<String[]> getTransactions() {

```

Comment:
Have you considered creating the `Transaction` objects here, and returning `List<Transaction>` instead? It would isolate all the parsing and error handling in this class, so `Bank` doesn't have to worry about interpreting the array of strings that you've read from the CSV file.

Have you got onto the next part of the exercise, where you read from a JSON file? How have you updated the code to handle that?